docs/build/autogenerate.rst

```diff
@@ -405,18 +407,24 @@ is set to True::
 .. note::
 
    The default type comparison logic (which is end-user extensible) currently
-   works for **major changes in type only**, such as between ``Numeric`` and
-   ``String``.     The logic will **not** detect changes such as:
+   works by comparing the generated SQL for a column. It does this in two steps-
 
-   * changes between types that have the same "type affinity", such as
-     between ``VARCHAR`` and ``TEXT``, or ``FLOAT`` and ``NUMERIC``
 
-   * changes between the arguments within the type, such as the lengths of
-     strings, precision values for numerics, the elements inside of an
-     enumeration.
+   * First, it compares the outer type of each column such as ``VARCHAR``
+     or ``TEXT``. Dialect implementations can have synonyms that are considered
+     equivalent- this is because some databases support types by converting them
+     to another type. For example, in ``NUMERIC`` and ``DECIMAL`` are considered
+     equivalent.
```

Comment:
**mike bayer** (zzzeek) wrote:

"for example, NUMERIC and DECIMAL are considered equivalent on all backends, while on the Oracle backend the additional synonyms BIGINT, INTEGER, NUMBER, SMALLINT are added to this list of equivalents"

Reply:
good one!